JenkinsFile

```diff
@@ -1,82 +1,238 @@
+                } else {
+                  git url: "https://github.com/venicegeo/pztest-integration.git", branch: "master"
+                }
+                sh "npm install newman@2"
+                sh "ci/blackbox.sh"
+              }
+            }
+          }
+        }
     }
+    if(params.DEPLOY_PHASE_TWO) {
+	  stage('Phase Two Deploy') {
+		if(!fileExists('.cf')) {
+		  sh "mkdir -p .cf"
+		}
+		withEnv(['CF_HOME=.cf']) {
+		  withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: "${env.PCF_CREDS}", usernameVariable: 'CFUSER', passwordVariable: 'CFPASS']]) {
+			sh """cf api ${env.PCF_API_ENDPOINT}
+			cf auth ${CFUSER} ${CFPASS}
+			cf target -o ${env.PCF_ORG} -s ${env.PHASE_TWO_PCF_SPACE}
+			cf push ${appName} --hostname ${appName} -f manifest.jenkins.yml -d ${env.PHASE_TWO_PCF_DOMAIN} --no-start
+			cf set-env ${appName} SPACE ${env.PHASE_TWO_PCF_SPACE}
+			cf set-env ${appName} DOMAIN ${env.PHASE_TWO_PCF_DOMAIN}
+			cf start ${appName}"""
+		  }
+		}
+		withCredentials([[$class: 'StringBinding', credentialsId: "${env.THREADFIX_API_KEY}", variable: 'THREADFIX_KEY']]) {
+		  sh """
+		   mkdir -p ${root}/zap-out
+		   ${zapHome}/zap.sh -cmd \
+			-quickout ${root}/zap-out/zap.xml \
+			-quickurl https://${appName}.${env.PHASE_TWO_PCF_DOMAIN}
+		   cat ${root}/zap-out/zap.xml
+		   /bin/curl -v --insecure -H 'Accept: application/json' -X POST --form file=@${root}/zap-out/zap.xml ${env.THREADFIX_URL}/rest/applications/${THREADFIX_ID}/upload?apiKey=${THREADFIX_KEY}
+		 """
+		}
+		withEnv(['CF_HOME=.cf']) {
+		  withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: "${env.PCF_CREDS}", usernameVariable: 'CFUSER', passwordVariable: 'CFPASS']]) {
+			sh """
+			 cf api ${env.PCF_API_ENDPOINT}
+			 cf auth ${CFUSER} ${CFPASS}
+			 cf target -o ${env.PCF_ORG} -s ${env.PHASE_TWO_PCF_SPACE}"""
+			def legacyAppNames = sh(script: "cf routes | grep \"pz-logger \" | awk '{print \$4}'", returnStdout: true)
+			sh "cf map-route ${appName} ${env.PHASE_TWO_PCF_DOMAIN} --hostname pz-logger"
+		    // Remove Legacy applications
+		    for (Object legacyApp : legacyAppNames.trim().tokenize(',')) {
+		      def legacyAppName = legacyApp.toString().trim()
+			  if (legacyAppName != appName) {
+		        sh "cf delete -f ${legacyAppName}"
+			  }
+		    }
+		  }
+		}
+	  }
   }
 
-  stage('Cleanup') {
-    deleteDir()
-  }
+    if (!params.SKIP_INTEGRATION_TESTS) {
+      stage('Phase Two Integration Tests') {
+        withCredentials([file(credentialsId: "${POSTMAN_SECRET_FILE}", variable: 'POSTMAN_FILE')]) {
+          withEnv(["PATH+=${root}/integration-tests/node_modules/newman/bin:${nodejs}/bin", "PCF_SPACE=${env.PHASE_TWO_PCF_SPACE}", "HOME=${WORKSPACE}"]) {
+            dir ("integration-tests") {
+              sh "ci/blackbox.sh"
+            }
+          }
+        }
+      }
+    }
```

Comment:
Same thing as int environment tests. If the integration tests don't test pz-logger at all, then we don't need these steps.